### PR TITLE
Host library fixes & update example host wallet app

### PR DESCRIPTION
**Pull request by vishnumad**
### _Summary_
* Host library fixes
* Adds example host wallet app

### _How did you test your changes?_
Manually

https://user-images.githubusercontent.com/3137711/203421626-2b854d62-ae69-42a3-beea-96b71ab2538c.mp4

### CODE REVIEW

File: react-native/host/src/events/events.ts [hunk 1]
```diff
@@ -108,8 +108,11 @@ type MWPEvent =
 
 const diagnosticLogger = new EventEmitter();
 
-export function addDiagnosticLogListener(listener: (event: MWPEvent) => void): EmitterSubscription {
-  return diagnosticLogger.addListener('mwp_event', listener);
+export function addDiagnosticLogListener(listener: (event: MWPEvent) => void) {
+  diagnosticLogger.addListener('mwp_event', listener);
+  return () => {
+    diagnosticLogger.removeListener('mwp_event', listener);
+  };
```
Comment (bangtoven) [inline react-native/host/src/events/events.ts:115]: 👍

File: react-native/host/src/provider/MobileWalletProtocolProvider.tsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ type MWPHostContextType = {
   handleRequestUrl: (url: string) => Promise<boolean>;
   fetchClientAppMetadata: () => Promise<AppMetadata | null>;
   isClientAppVerified: () => Promise<boolean>;
-  approveHandshake: (action: HandshakeAction, metadata: AppMetadata) => Promise<void>;
+  approveHandshake: (metadata: AppMetadata | null) => Promise<void>;
```
Comment (bangtoven) [inline react-native/host/src/provider/MobileWalletProtocolProvider.tsx:37]: not a js/ts experts, but afaik, we'd better use `undefined` unless `null` has a specific meaning on the context. https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined  

how about updating like
```suggestion
  approveHandshake: (metadata: AppMetadata | undefined) => Promise<void>;
```
and line 35 to return the same type as well.

**Review by bangtoven — COMMENTED** [review]
@vishnumad

File: react-native/host/src/provider/MobileWalletProtocolProvider.tsx (hunk 2 above)
Comment (vishnumad) [reply react-native/host/src/provider/MobileWalletProtocolProvider.tsx:37]: @bangtoven, I wanted it to be intentional on the consumer's side not to pass metadata in. With undefined, it's easy to forget to pass in the param and just call `approveHandshake()`. The `fetchAppMetadata` function also returns a `AppMetadata | null`, so this works out well.

**Review by bangtoven — APPROVED** [review]

**Pull request merged**